# Analyzing-RNA-seq-data-with-DESeq2
Analyzing count data from RNA-seq involves a fundamental task: identifying differentially expressed genes. The data is structured as a table, detailing the number of sequence fragments assigned to each gene for every sample. Similar data formats emerge in various assay types such as comparative ChIP-Seq, HiC, shRNA screening, and mass spectrometry. A key analytical objective is to quantify and statistically infer systematic changes between conditions, relative to within-condition variability.

DESeq2 is a package that addresses this challenge, employing negative binomial generalized linear models to test for differential expression. The estimates of dispersion and logarithmic fold changes incorporate data-driven prior distributions. This vignette serves to elucidate the usage of the DESeq2 package, providing insights into typical workflows. While a Bioconductor website RNA-seq workflow covers similar content at a more deliberate pace, including count matrix generation from FASTQ files, this document specifically focuses on DESeq2 package version 1.43.0.
